sdk/src/main/scala/com/horizen/forge/Forger.scala

```diff
@@ -33,16 +33,16 @@ class Forger(settings: SidechainSettings,
   @volatile private var forgingIsActive: Boolean = false
 
   val consensusMillisecondsInSlot: Int = params.consensusSecondsInSlot * 1000
-  val forgingInitiator: Runnable = () => {
-    while (true) {
+  val forgingInitiator: TimerTask = new TimerTask {
+    override def run(): Unit = {
```

Comment:
remove ` @volatile private var forgingIsActive` and assign timer to some `val timer`.
So, in startForging, just `cancel` timer and schedule it again with `.scheduleAtFixedRate(forgingInitiator, 0, consensusMillisecondsInSlot)`. In stopForging just `cancel` the timer.